cleanlab/object_detection/summary.py

```diff
@@ -444,6 +450,163 @@ def visualize(
+    num_images = len(predictions)
+    pool = Pool(num_procs)
+    counter_dict: DefaultDict[int, dict[str, int]] = collections.defaultdict(
+        lambda: {"TP": 0, "FP": 0, "FN": 0}
+    )
+
+    for class_num in range(num_classes):
+        pred_bboxes, lab_bboxes = _filter_by_class(labels, predictions, class_num)
+        tpfpfn = pool.starmap(
+            _calculate_true_positives_false_positives,
+            zip(
+                pred_bboxes,
+                lab_bboxes,
+                [iou_threshold for _ in range(num_images)],
+                [True for _ in range(num_images)],
+            ),
+        )
+
+        for image_idx, (tp, fp, fn) in enumerate(tpfpfn):  # type: ignore
+            counter_dict[class_num]["TP"] += np.sum(tp)
+            counter_dict[class_num]["FP"] += np.sum(fp)
+            counter_dict[class_num]["FN"] += np.sum(fn)
+
+    return counter_dict
+
+
+def get_average_per_class_confusion_matrix(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+    num_procs: int = 1,
+    class_names: Optional[List[Union[int, str]]] = None,
+) -> Dict[Union[int, str], Dict[str, float]]:
+    """
+    Compute a confusion matrix dictionary for each class containing the average number of True Positive, False Positive, and False Negative detections from the object detection model across a range of Intersection over Union thresholds.
+
+    The confusion matrices can help you identify model strengths and determine potential biases.
+
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`object_detection.filter.find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    predictions:
+        A list of ``N`` ``np.ndarray`` such that ``predictions[i]`` corresponds to the model predictions for the `i`-th image.
+        Refer to documentation for this argument in :py:func:`object_detection.filter.find_label_issues <cleanlab.object_detection.filter.find_label_issues>` for further details.
+
+    num_procs:
+        Number of processes for parallelization. Default is 1.
+
+    class_names:
+        Optional dictionary mapping one-hot-encoded class labels back to their original class names in the format ``{"integer-label": "original-class-name"}``
+
+
+    Returns
+    -------
+    avg_metrics: dict
+        A distionary containing the average confusion matrix.
+
+        The default range of Intersection over Union thresholds is from 0.5 to 0.95 with a step size of 0.05.
+    """
+    iou_thrs = np.linspace(0.5, 0.95, int(np.round((0.95 - 0.5) / 0.05)) + 1, endpoint=True)
+    num_classes = len(predictions[0])
+    if class_names is None:
+        class_names = list(range(num_classes))
+    avg_metrics = {class_num: {"TP": 0.0, "FP": 0.0, "FN": 0.0} for class_num in class_names}
```

Comment:
I get KeyError when trying to run 
```python
class_names = {"0": "car", "1": "hand", "2": "bowl", "3": "human", "4": "traffic light"}
get_average_per_class_confusion_matrix(labels=labels,predictions=predictions,num_procs=1,class_names=class_names)
```

Reason seems to be avg_metrics is initialized as dict of {class_num: {}} while access in loop is `avg_metrics[class_names[class_num]]["FP"] += fp`.  I would suggest initializing avg_metrics as class_names values instead of keys if class_names is passed in.

Reply:
It's changed to be sorted by indices, should be fine now